src/bin/util/softhsm2-util-ossl.cpp

```
@@ -1139,3 +1168,198 @@ void crypto_free_eddsa(eddsa_key_material_t* keyMat)
+		return NULL;
+	}
+
+	uint8_t seed[32];
+	size_t seed_len;
+	int rv = EVP_PKEY_get_octet_string_param(pkey, OSSL_PKEY_PARAM_ML_DSA_SEED,
+								seed, sizeof(seed), &seed_len);
+
+	if(!rv) {
+		fprintf(stderr, "ERROR: Could not get ML-DSA seed, rv: %d", rv);
+		memset(seed, 0, sizeof(seed));
+		return NULL;
```

Cppcheck (2.20.0)

[error] Memory leak

(memleak)